overlord/devicestate/firstboot_test.go

```diff
@@ -58,75 +59,82 @@ import (
 
 	overlord *overlord.Overlord
 
 	perfTimings timings.Measurer
 }
 
-var _ = Suite(&FirstBootTestSuite{})
-
-func (s *FirstBootTestSuite) SetUpTest(c *C) {
-	s.BaseTest.SetUpTest(c)
+func (t *firstBootBaseTest) setupBaseTest(c *C, s *seedtest.SeedSnaps) {
+	t.BaseTest.SetUpTest(c)
 
 	tempdir := c.MkDir()
 	dirs.SetRootDir(tempdir)
-	s.AddCleanup(func() { dirs.SetRootDir("/") })
+	t.AddCleanup(func() { dirs.SetRootDir("/") })
 
-	s.AddCleanup(release.MockOnClassic(false))
+	t.AddCleanup(release.MockOnClassic(false))
 
 	// mock the world!
 	err := os.MkdirAll(filepath.Join(dirs.SnapSeedDir, "snaps"), 0755)
 	c.Assert(err, IsNil)
-	err = os.MkdirAll(filepath.Join(dirs.SnapSeedDir, "assertions"), 0755)
-	c.Assert(err, IsNil)
 
 	err = os.MkdirAll(dirs.SnapServicesDir, 0755)
 	c.Assert(err, IsNil)
 	os.Setenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS", "1")
-	s.AddCleanup(func() { os.Unsetenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS") })
-	s.systemctl = testutil.MockCommand(c, "systemctl", "")
-	s.AddCleanup(s.systemctl.Restore)
-
-	err = ioutil.WriteFile(filepath.Join(dirs.SnapSeedDir, "seed.yaml"), nil, 0644)
-	c.Assert(err, IsNil)
+	t.AddCleanup(func() { os.Unsetenv("SNAPPY_SQUASHFS_UNPACK_FOR_TESTS") })
+	t.systemctl = testutil.MockCommand(c, "systemctl", "")
+	t.AddCleanup(t.systemctl.Restore)
 
-	s.TestingSeed = &seedtest.TestingSeed{}
-	s.SetupAssertSigning("can0nical", s)
+	s.SetupAssertSigning("canonical")
 	s.Brands.Register("my-brand", brandPrivKey, map[string]interface{}{
 		"verification": "verified",
 	})
 
-	s.SnapsDir = filepath.Join(dirs.SnapSeedDir, "snaps")
-	s.AssertsDir = filepath.Join(dirs.SnapSeedDir, "assertions")
-
-	s.devAcct = assertstest.NewAccount(s.StoreSigning, "developer", map[string]interface{}{
+	t.devAcct = assertstest.NewAccount(s.StoreSigning, "developer", map[string]interface{}{
 		"account-id": "developerid",
 	}, "")
 
-	s.AddCleanup(ifacestate.MockSecurityBackends(nil))
+	t.AddCleanup(sysdb.InjectTrusted([]asserts.Assertion{s.StoreSigning.TrustedKey}))
+	t.AddCleanup(ifacestate.MockSecurityBackends(nil))
 
 	ovld, err := overlord.New(nil)
 	c.Assert(err, IsNil)
 	ovld.InterfaceManager().DisableUDevMonitor()
-	s.overlord = ovld
+	t.overlord = ovld
 	c.Assert(ovld.StartUp(), IsNil)
 
 	// don't actually try to talk to the store on snapstate.Ensure
 	// needs doing after the call to devicestate.Manager (which happens in overlord.New)
 	snapstate.CanAutoRefresh = nil
 
-	s.perfTimings = timings.New(nil)
+	t.perfTimings = timings.New(nil)
+}
+
+type firstBoot16Suite struct {
+	firstBootBaseTest
+
+	// TestingSeed16 helps populating seeds (it provides
+	// MakeAssertedSnap, WriteAssertions etc.) for tests.
+	*seedtest.TestingSeed16
+}
+
+var _ = Suite(&firstBoot16Suite{})
+
+func (s *firstBoot16Suite) SetUpTest(c *C) {
+	s.TestingSeed16 = &seedtest.TestingSeed16{}
+
+	s.setupBaseTest(c, &s.TestingSeed16.SeedSnaps)
+
+	s.SeedDir = dirs.SnapSeedDir
+
+	err := os.MkdirAll(filepath.Join(dirs.SnapSeedDir, "assertions"), 0755)
```

Comment:
ah so we create the assertions dir here instead of in the setup for firstBootBaseTest... Hmm is this because we don't really need the assertions dir in the (eventual) firstBoot20Suite tests?

Reply:
correct, the is a top level <system-seed>/snaps directory in Core 20 seed/recovery partitions, but not anymore a top level /assertions one.